transformation/plugins/org.jboss.tools.fuse.transformation.editor/src/main/java/org/jboss/tools/fuse/transformation/editor/internal/wizards/Hl7Page.java

```diff
@@ -0,0 +1,570 @@
+
+        hl7PreviewText = new Text(group2, SWT.V_SCROLL | SWT.READ_ONLY | SWT.H_SCROLL );
+        hl7PreviewText.setBackground(_page.getBackground());
+
+        bindControls();
+        validatePage();
+    }
+
+    private String getJsonText(IFile jsonFile) {
+        if (jsonFile != null && jsonFile.exists()) {
+            try (final InputStream istream = jsonFile.getContents()) {
+                final StringBuffer buffer = new StringBuffer();
+                try (final BufferedReader in = new BufferedReader(new InputStreamReader(istream))) {
+                    String inputLine;
+                    while ((inputLine = in.readLine()) != null) {
+                        buffer.append(inputLine + "\n");
+                    }
+                }
+                return buffer.toString();
+            } catch (final CoreException e1) {
+                e1.printStackTrace();
+            } catch (final IOException e1) {
+                e1.printStackTrace();
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public boolean isSourcePage() {
+        return isSource;
+    }
+
+    @Override
+    public boolean isTargetPage() {
+        return !isSource;
+    }
+
+    boolean fileIsEmpty(final String path) {
+        try {
+            IFile testIFile = model.getProject().getFile(path);
+            if (testIFile.exists()) {
+                File testFile = testIFile.getRawLocation().makeAbsolute().toFile();
+                if (testFile.length() == 0) {
+                    return true;
+                }
+                String jsonText = getJsonText(testIFile);
+                if (jsonText != null && jsonText.trim().isEmpty()) {
+                    return true;
+                }
+            } else {
+                return true;
+            }
+        } catch (Exception e) {
+            // ignore
+        }
+        return false;
+    }
+
+    boolean jsonSchema(final String path) throws Exception {
+        final IPath tempPath = new Path(path);
+        final IFile jsonFile = model.getProject().getFile(tempPath);
+        if (fileIsEmpty(path)) {
+            return false;
+        }
+        File testFile = jsonFile.getRawLocation().makeAbsolute().toFile();
+        try (FileInputStream fileInput = new FileInputStream(testFile)) {
+            char quote = '\0';
+            final StringBuilder builder = new StringBuilder();
+            int r;
+            while ((r = fileInput.read()) != -1) {
+                char chr = (char) r;
+                // Find quote
+                if (quote == '\0') {
+                    if (chr == '"' || chr == '\'') {
+                        quote = chr;
+                    }
+                } else if (chr == quote) {
+                    final String keyword = builder.toString();
+                    switch (keyword) {
+                    case "$schema":
+                    case "title":
```

Comment:
no break?
so we can just check that the "id".equals(keyword), or I miss something?